pkg/controller/core/workload_controller.go

```diff
@@ -337,24 +335,24 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	if workload.HasQuotaReservation(&wl) {
 		if evictionTriggered, err := r.reconcileCheckBasedEviction(ctx, &wl); evictionTriggered || err != nil {
-			return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
Why you reverted this?

Reply:
This part of the code had no issues before, so it has been temporarily reverted.

Reply:
Reverting client.IgnoreNotFound(err) back here shouldn't cause any issues.

Reply:
+1 on @mbobrovskyi

Reply:
Are the following steps really needed? 

If so it is concerning it is so easy to break

Is it feasible to add a unit or integration test to verify the change causes a bug. It could be a follow up Issue or PR

Reply:
> Are the following steps really needed?
> 
> If so it is concerning it is so easy to break
> 
> Is it feasible to add a unit or integration test to verify the change causes a bug. It could be a follow up Issue or PR

This change has already reverted.

Reply:
Yes, this change was indeed not covered by unit test cases. Let's add some unit tests here later.